torch/distributed/tensor/parallel/style.py

```diff
@@ -1,652 +1,357 @@
-    """
-    _deprecate_warnings("make_sharded_output_tensor", "Specify output_layouts instead.")
-    return output.to_local()  # type: ignore[call-arg]
+    def apply(self, module: nn.Module, device_mesh: DeviceMesh) -> nn.Module:
+        ...
 
 
-@_prepare_output_validate  # type: ignore[arg-type] # pyre-ignore[56]
-def make_output_reshard_tensor(
-    output: DTensor,
-    device_mesh: Optional[DeviceMesh] = None,
-) -> torch.Tensor:  # noqa: D205, D400
-    """
-    .. warning::
-        This method was deprecated and please specify ``output_layouts`` instead.
-    """
-    _deprecate_warnings("make_output_reshard_tensor", "Specify output_layouts instead.")
-    return make_output_shard_1d(output, device_mesh).to_local()  # type: ignore[call-arg, attr-defined, misc]
-
-
-def _redistribute_per_both_layouts(t, input_layout, output_layout, device_mesh):
-    dst_placements = (output_layout,)
-    if isinstance(t, DTensor):
-        return t.redistribute(device_mesh, dst_placements)
-    elif isinstance(t, torch.Tensor):
-        dtensor = DTensor.from_local(t, device_mesh, (input_layout,), run_check=False)
-        return dtensor.redistribute(device_mesh, dst_placements)
-    else:
-        if input_layout is not None:
-            raise RuntimeError(
-                "Tensor parallel module expects DTensor or tensor"
-                f" when layout specified but received {type(t)}!"
-            )
-        else:
-            return t
-
-
-def _redistribute_per_layout(layout, use_local_output, t, device_mesh):
-    dst_placements = (layout,)
-    if isinstance(t, DTensor):
-        dtensor = t.redistribute(device_mesh, dst_placements)
-        return dtensor.to_local() if use_local_output else dtensor
-    else:
-        if layout is not None:
-            raise RuntimeError(
-                "Tensor parallel module expects DTensor or tensor"
-                f" when layout specified but received {type(t)}!"
-            )
-        else:
-            return t
-
-
-class RowwiseParallel(ParallelStyle):
+class ColwiseParallel(ParallelStyle):
     """
-    Partition the row of a module.
-
-    We assume the input to be a sharded :class:`DTensor` and output to be a :class:`torch.Tensor`.
-
-    Args:
-        input_layouts (Union[Placement, Tuple[Placement, ...]]):
-            The layout of input tensor(s) which DTensor will be created upon.
-        output_layouts (Union[Placement, Tuple[Placement, ...]]):
-            The layout of input tensor(s) which created DTensor will be redistributed to.
-        use_local_output (bool):
-            Whether to convert the DTensor to local :class:`torch.Tensor`.
-
+    Partition the column of a nn.Module and allow the parallelized module to run colwise sharded computation.
+    Users can compose it with RowwiseParallel to achieve the sharding of more complicated modules.
+
+    Keyword Args:
+        input_layouts (Placement, optional):
```

Comment:
Would it be possible to call this arg `input_placement` (singular) (and similarly `output_placement`) since the type is of `Placement`? Is the concern backward compatibility?

(Or, if we want consistency with `PrepareModuleInput`, whose `input_layouts` accepts `Tuple[Placement, ...]`, then we can name them all `input_placements` (plural)?)

Reply:
`input_placements` is sth I've considered, the reason I ended up not using them is that I would like to avoid user confusion when they directly using DTensor APIs, `placements` is a specific term to DTensor where it's a List/Tuple of placement that "corresponding to Device Mesh dimensions". However here for `PrepareModuleInput/Output`, it also have a Tuple of Placement, but it corresponds to "multiple tensors/inputs of the  nn.Module", so user might be confused if the naming are the same.. lmk if this make sense

Reply:
Ohh, I see. That makes sense! Thanks for the explanation!